modules/apps/portal-security/portal-security-auto-login/src/main/java/com/liferay/portal/security/auto/login/remember/me/RememberMeAutoLogin.java

```diff
@@ -78,21 +81,22 @@ protected String[] doLogin(
 
 		String[] credentials = null;
 
-		if (Validator.isNotNull(autoUserId) &&
-			Validator.isNotNull(autoPassword) &&
-			Validator.isNotNull(rememberMe)) {
+		if (Validator.isNotNull(rememberMe) &&
+			Validator.isNotNull(rememberMeTokenId) &&
+			Validator.isNotNull(rememberMeTokenToken)) {
 
 			Company company = _portal.getCompany(httpServletRequest);
 
 			if (company.isAutoLogin()) {
-				KeyValuePair kvp = _userLocalService.decryptUserId(
-					company.getCompanyId(), autoUserId, autoPassword);
+				KeyValuePair kvp = _rememberMeTokenLocalService.validateToken(
+					GetterUtil.getLong(rememberMeTokenId),
+					rememberMeTokenToken);
```

Comment:
Did you find other code where validate* methods has a return?

Reply:
When looking for it, I found just some rare cases like: https://github.com/liferay/liferay-portal/blob/d0fde45a6664d8b78a8644c9e1e8f4fa995746e0/modules/apps/account/account-service/src/main/java/com/liferay/account/service/impl/AccountEntryLocalServiceImpl.java#L1185. But in general, it's either void or boolean, so I think we shouldn't indeed name this validate... 

I though that we could refactor it and call something like getUserCredentials() and we could simply directly return the credentials[] string instead of the KeyValuePair. What do you think? Do you have a cooler suggestion? I always find it hard to name things :laughing: